src/rebase.js

```diff
@@ -116,13 +116,22 @@ module.exports = (function(){
   function _fetch(endpoint, options){
     _validateEndpoint(endpoint);
     optionValidators.context(options);
-    optionValidators.then(options);
     options.queries && optionValidators.query(options);
     var ref = firebase.database().ref(endpoint);
     ref = _addQueries(ref, options.queries);
-    ref.once('value', (snapshot) => {
+    return ref.once('value').then(snapshot => {
       var data = options.asArray === true ? _toArray(snapshot) : snapshot.val();
-      options.then.call(options.context, data);
+      if(options.then){
+        options.then.call(options.context, data);
+      }
+      return data;
+    }, err => {
+      //call onFailure callback if it exists otherwise return a rejected promise
+      if(options.onFailure && typeof options.onFailure === 'function'){
+        options.onFailure.call(options.context, err);
+      } else {
```

Comment:
@anotherpit I don't think it can be without forcing you to use the Promise. Like if you use the onFailure it will still throw an unhandled rejection I think is why I did that.